### DISCUSSION TITLE
do I need to re-train models when installing a hot fix?

### DISCUSSION THREAD

**Discussion by zliendo** [autogluon/autogluon#2289] [Q&A] [answered]
regarding
"Version 0.5.2
v0.5.2 is a security hotfix release.
This release is non-breaking when upgrading from v0.5.0. As always, only load previously trained models using the same version of AutoGluon that they were originally trained on. Loading models trained in different versions of AutoGluon is not supported."

if I upgrade to a new autogluon hotfix version, e.g; to 0.5.2 from 0.5.0, 
do I need to re-train models built with 0.5.0?

**Comment by gradientsky** [comment, marked as answer]
If you are using autogluon.multimodal, then there were some updates in 0.5.1 which might be helpful if retraining performed: see the [release notes for 0.5.1](https://github.com/awslabs/autogluon/releases/tag/v0.5.1). For tabular I don't think there are any functional changes beyond libraries updates related to security, so it should work without retraining.

**Reply by Innixma** [reply]
Note that loading with a different version, even from v0.5.0 to v0.5.2, is not recommended, even if it works. I have no idea if it will work, and we make no attempt at ensuring it works. I personally highly doubt it will work.

**Reply by zliendo** [reply]
Thank you so much for your prompt replies!

And this is not just  because the models are being persisted in pickle format, correct? 
iow, If we had the models in onyx format, still re-training would be recommended when installing hot-fixes?

**Reply by alne4294** [reply]
@Innixma can you speak to @zliendo 's question?   Wondering why (particularly in the context of vulnerability management), we don't attempt to create _some_ backwards compatible patches?

**Reply by Innixma** [reply]
I don't attempt to ensure backward compatibility because the overall system is too complex to test and maintain backward compatibility. Doing so would slow down our development velocity, significantly raise the bar to entry for contributors, and only assist a minority of users. If backwards compatibility is important for you, you can try to load the new version with your old models and see if it works, but we do not provide debugging support for this use-case. Alternatively, you can apply the security patches yourself to the old version of AutoGluon and test that it works with your models, but I wouldn't recommend this due to the technical complexity.
